### PR TITLE
feat(ux): show syntax-highlighted SQL if pygments is installed

### PR DESCRIPTION
**Pull request by cpcloud**
Adds syntax-highlighted output in the IPython console.

image

### CODE REVIEW

**Comment by cpcloud** [comment]
Slightly gnarlier example using BigQuery.

For this ibis code:

image

We get this SQL:

image

File: ibis/expr/sql.py [hunk 1]
```diff
@@ -336,6 +336,23 @@ def __repr__(self) -> str:
...
 
+    def _repr_pretty_(self, p, cycle) -> str:
+        try:
+            from pygments import highlight
+            from pygments.formatters import TerminalFormatter
+            from pygments.lexers import SqlLexer
+        except ImportError:
+            output = str(self)
+        else:
+            output = highlight(
```
Comment (jcrist) [inline ibis/expr/sql.py:347]: Is there a chance that highlighting can error for any of the dialects we support?

Also, these lines appear uncovered, do we want to add a test for this?

**Review by jcrist — APPROVED** [review]
Nice! Just a question, otherwise LGTM

File: ibis/expr/sql.py (hunk 1 above)
Comment (cpcloud) [reply ibis/expr/sql.py:347]: Possibly, but it's hard to say.

Picking an arbitrary example not common to most sql dialects (duckdb's struct literal syntax), highlighting seems to work:

image

I can catch `Exception` and warn in the case anything goes wrong (or maybe just avoid warning?)

Comment (cpcloud) [reply ibis/expr/sql.py]: Added a `contextlib.suppress` around the `highlight` call.

**Pull request merged**